piker/brokers/ib.py

```diff
@@ -1183,112 +1201,217 @@ def pack_position(pos: Position) -> Dict[str, Any]:
     else:
         symbol = con.symbol
 
-    return {
-        'broker': 'ib',
-        'account': pos.account,
-        'symbol': symbol,
-        'currency': con.currency,
-        'size': float(pos.position),
-        'avg_price': float(pos.avgCost) / float(con.multiplier or 1.0),
-    }
+    return BrokerdPosition(
+        broker='ib',
+        account=pos.account,
+        symbol=symbol,
+        currency=con.currency,
+        size=float(pos.position),
+        avg_price=float(pos.avgCost) / float(con.multiplier or 1.0),
+    )
 
 
-@tractor.msg.pub(
-    send_on_connect={'local_trades': 'start'}
-)
-async def stream_trades(
+async def handle_order_requests(
```

Comment:
So basically broker integration to the paper engine is setting up two stream consumers, one for paper orders placed on `piker`'s UI, and another one for the broker order flow.

Really good!

Reply:
Not the *paper engine* per say, but the *clearing engine* (aka `emsd` service), yes.

There's a single endpoint `trades_dialogue()` which is passed a `tractor.Context`, this endpoint then opens a bidir `tractor` stream and starts handling inbound order requests, which this `handle_order_requests()` *task* uses to make calls to place orders on the backend. The main entrypoint, `trades_dialogue()` (or any other task you'd like) must ship back trade status messages based on updates from the target broker.